File: Data Structures & Algorithms/reorder-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {void}
     */
    reorderList(head) {
        let slow = head;
        let fast = head;

        while(fast && fast.next) {
            slow = slow.next;
            fast = fast.next.next;
        }

        // Now slow is in the middle of the list
        // and fast is at the end of the list

        let firstList = head;
        let secondList = slow.next;
        slow.next = null;

        // Reverse second list

        let prev = null;
        while(secondList !== null) {
            let temp = secondList.next;

            secondList.next = prev;
            prev = secondList;
            secondList = temp;
        }

        secondList = prev;

        while(secondList !== null) {
            const temp1 = firstList.next;
            const temp2 = secondList.next;

            firstList.next = secondList;
            secondList.next = temp1;

            firstList = temp1;
            secondList = temp2;
        }
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        sum1 = 0 
        sum2 = 0
        n = 0
        while l1:
            sum1 += l1.val * pow(10, n)
            l1 = l1.next
            n += 1
        
        n = 0
        while l2:
            sum2 += l2.val * pow(10, n)
            l2 = l2.next
            n += 1
        
        resultSum = sum1 + sum2

        num = resultSum % 10
        curr = head = ListNode(num)
        resultSum = resultSum // 10

        while resultSum:
            num = resultSum % 10
            curr.next = ListNode(num)
            resultSum = resultSum // 10
            curr = curr.next


        return head


        

        